File: 代码随想录/valid-perfect-square/res.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        int l = 0 , r = num;
        while (l < r)
        {
            int mid = (l + r + 1ll) / 2;
            if (mid <= num / mid ) l = mid;
            else r = mid - 1;
        }
        if (l * l == num) return true;
        else return false;
    }
};

File: 代码随想录/valid-perfect-square/res.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function(num) {
    let l = 0, r = num;
    while (l < r)
    {
        let mid = l + r + 1 >> 1;
        if (mid * mid <= num) l = mid;
        else r = mid - 1;
    }
    if (l * l == num) return true;
    else return false;
};